src/ripple/app/hook/Enum.h

```diff
@@ -372,105 +376,129 @@ const double fee_base_multiplier = 1.1f;
 
 #define HOOK_WRAP_PARAMS(...) __VA_ARGS__
 #define HOOK_API_DEFINITION(RETURN_TYPE, FUNCTION_NAME, PARAMS_TUPLE) \
-    {                                                                 \
-#FUNCTION_NAME,                                               \
-        {                                                             \
-            RETURN_TYPE, HOOK_WRAP_PARAMS PARAMS_TUPLE                \
-        }                                                             \
-    }
+    whitelist[#FUNCTION_NAME] = {RETURN_TYPE, HOOK_WRAP_PARAMS PARAMS_TUPLE};
 
 using APIWhitelist = std::map<std::string, std::vector<uint8_t>>;
 
 // RH NOTE: Find descriptions of api functions in ./impl/applyHook.cpp and
 // hookapi.h (include for hooks) this is a map of the api name to its return
 // code (vec[0] and its parameters vec[>0]) as wasm type codes
-static const APIWhitelist import_whitelist{
+inline APIWhitelist
+getImportWhitelist(
+#ifndef GUARD_CHECKER_BUILD
+    Rules const& rules
+#endif
```

Comment:
I feel like leaving the param in and making it a std::optional is a bit cleaner than changing the definition based on how its compiled?